src/node/referee.js

```diff
@@ -0,0 +1,38 @@
+const gameStatus = require('./gameStatus.js');
+const checkWin = require('./checkWin.js');
+const checkDraw = require('./checkDraw.js');
+
+function Referee() {
+  const referee = {
+    gameStatus: gameStatus.ONGOING,
+    getGameStatus: () => {
+      return referee.gameStatus;
+    },
+    updateGameStatus: (userMove, game) => {
+      if (checkWin(userMove, game.board)) {
+        referee.gameStatus = gameStatus.WIN;
+        game.board.print();
```

Comment:
I think it's a good time to talk about side-effects! This is a term you'll come across a lot in JS, but it's not restricted to the language. Pay attention to function that are doing more than the function name tells you about!
`updateGameStatus` appears to manipulate only the internal `gameStatus` flag. But it does more than that! It Prints the board and announce winner/draw.
That's not actually a problem, but you closed your program to extension. Now, this Referee can only "printBoard" and "console.log" your message. But what if I wanted to keep doing that to tic-tac-toe game and in another type of game, have different things I want to do when game in won/draw?
One option would be introducing a "gameType" to your referee, and if/else to oblivion. 
``` js
updateGameStatus: (userMove, game) => {
      if(referee.gameType === GAME_TYPES.TIC_TAC_TOE) {
           if (checkWin(userMove, game.board)) {
                 referee.gameStatus = gameStatus.WIN;
                 game.board.print();
                ....
           }
      }
          if(referee.gameType === GAME_TYPES.CHECKERS) {
           if (checkWin(userMove, game.board)) {
                 referee.gameStatus = gameStatus.WIN;
                 game.board.print();
                ....
           }
      }
}
```
Not quite good, because if another dev, responsible for the other type of game, came here to update it's own usecase, (s)he might break your code, simply because they are physically near each other!
You could achieve this flexibility using Object Oriented class inheritence. A base Referee class and inherited TicTacToeReferee and CheckersReferee. That would isolate better both use cases, BUT still has the danger of whoever touches base `Referee` to break every other N inherited classes (real caveat of OO, and that one of main reasons I consider it broken)

What I would to add more flexibility to it, would be inject the outside world conditions to it, and only allow things related to a generic referee to live inside it!:

``` js
function Referee(rules, onMoveVictory, onMoveDraw) {
  const referee = {
    gameStatus: GAME_STATUS.ON_GOING,
    updateGameStatus() {
      if(rules.hasWin()) {
        referee.gameStatus = GAME_STATUS.WIN;
        onVictory();
      }
      if(rules.draw()) {
        referee.gameStatus = GAME_STATUS.DRAW;
        onMoveDraw();
      }
    }
  };
  return referee;
}
  const handleUserMakeWinningMove = () => {
    console.log('Contratulations!');
  }
  const handleUserMakeWinningMove = () => {
    console.log('Contratulations!');
  }

  const referee = Referee(ticTacToe, handleUserWinningMove, handleUserDrawMove)
```
This is an example of envisioning "what will most likely change in this code?", which is not always sane and predictable!

Reply:
TL;DR;
- Careful when making specific code for apparently generic entities
- Inject things you want flexibility over. I will need different set of rules? I will need to do different things when game is over? Pass them as constructor variables!